src/types.rs

```diff
@@ -221,12 +243,15 @@ impl From<&[Scalar]> for Label {
     fn from(v: &[Scalar]) -> Self {
         let mut res: Vec<u8> = vec![];
         v.iter()
-            .for_each(|val| res.append(&mut val.to_bytes().to_vec()));
+            .for_each(|val| res.append(&mut val.to_bytes().into()));
         Label(res)
     }
 }
-impl AsRef<[u8]> for Label {
-    fn as_ref(&self) -> &[u8] {
-        &self.0
+
+impl Deref for Label {
+    type Target = [u8];
+
+    fn deref(&self) -> &Self::Target {
+        self.0.as_slice()
```

Comment:
it's more often a question of style, but you can use `&self.0`. It is also slightly more generic, but to be honest nothing to worry here

Reply:
ok